src/client.ts

```diff
@@ -6806,7 +6806,7 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
      * Get the access token associated with this account.
      * @return {?String} The access_token or null
      */
-    public getAccessToken(): string {
+    public getAccessToken(): string | undefined {
         return this.http.opts.accessToken || null;
```

Comment:
These lines contradict each other, undefined != null

Reply:
good point! still learning typescript. 
a brief tour of the docs says
1) undefined is a type, null is an object,  https://stackoverflow.com/questions/5076944/what-is-the-difference-between-null-and-undefined-in-javascript Declaring the type to be undefined and then returning null makes sense to me but it's wrong! and will fail type checking in later versions of typescript

2) we should probably use undefined everywhere https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined but we're not, and I see the utility of null

3) there are a couple more places we do this, i'll update this pr to use `| null` and fix the rest